pkg/infra/httpclient/httpclientprovider/forwarded_cookies_middleware.go

```diff
@@ -0,0 +1,21 @@
+package httpclientprovider
+
+import (
+	"net/http"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend/httpclient"
+	"github.com/grafana/grafana/pkg/util/proxyutil"
+)
+
+const ForwardedCookiesMiddlewareName = "forwarded-cookies"
+
+// ForwardedCookiesMiddleware middleware that sets Cookie header on the
+// outgoing request, if forwarded cookies configured/provided.
+func ForwardedCookiesMiddleware(allowedCookies []string) httpclient.Middleware {
+	return httpclient.NamedMiddlewareFunc(ForwardedCookiesMiddlewareName, func(opts httpclient.Options, next http.RoundTripper) http.RoundTripper {
+		return httpclient.RoundTripperFunc(func(req *http.Request) (*http.Response, error) {
```

Comment:
Nit, maybe `return next` if allowedCookies are empty

Reply:
@marefr i had the same thought, and if i remember correctly returning next would forward all cookies instead of none

Reply:
Ohh than this doesn't work as expected. The incoming request cookies are never set on the outgoing request if not mistaken - probably need to provide the actual cookies to set or both incoming request Cookie header and allowedCookies. A bit weird

Reply:
i think this should be resolved with [fc89762](https://github.com/grafana/grafana/pull/50466/commits/fc89762249d8cf7bcb4e9220ef063c5aaf8ccfdb)